cfgov/v1/serializers.py

```diff
@@ -0,0 +1,87 @@
+from django.conf import settings
+
+from rest_framework import serializers
+
+from v1.jinja2tags.images import image_alt_value, wagtail_image_fn
+from v1.models import BlogPage, CFGOVPageCategory, EventPage
+from v1.util.ref import get_category_icon, is_blog, is_report
+
+
+class FilterPageSerializer(serializers.Serializer):
+    authors = serializers.ListField(source="get_authors")
+    categories = serializers.SerializerMethodField()
+    language = serializers.ChoiceField(choices=settings.LANGUAGES)
+    start_date = serializers.SerializerMethodField()
+    title = serializers.SerializerMethodField()
+    url = serializers.SerializerMethodField()
+    is_blog = serializers.SerializerMethodField()
+    is_event = serializers.SerializerMethodField()
+    is_report = serializers.SerializerMethodField()
+    event_location_str = serializers.SerializerMethodField()
+    tags = serializers.SerializerMethodField()
+    image_url = serializers.SerializerMethodField()
+    image_alt = serializers.SerializerMethodField()
+
+    def get_categories(self, page):
+        category_mapping = dict(CFGOVPageCategory.name.field.flatchoices)
+
+        categories = []
+        for category in page.categories.all():
+            category_slug = category.name
+            category_name = category_mapping[category_slug]
+            categories.append(
+                {
+                    "slug": category_slug,
+                    "name": category_name,
+                    "icon": get_category_icon(category_name),
+                }
+            )
+
+        return categories
+
+    def get_start_date(self, page):
+        return getattr(page, page.start_date_field)
+
+    def get_title(self, page):
+        return page.seo_title or page.title
+
+    def get_url(self, page):
+        return page.get_url(request=self.context.get("request"))
+
+    def get_is_blog(self, page):
+        return isinstance(page, BlogPage) or is_blog(page)
+
+    def get_is_event(self, page):
+        return isinstance(page, EventPage)
+
+    def get_is_report(self, page):
+        return is_report(page)
+
+    def get_event_location_str(self, page):
+        return page.location_str if isinstance(page, EventPage) else None
+
+    def get_tags(self, page):
+        return [
+            {
+                "slug": tag.slug,
+                "text": tag.name,
+                "url": f"?topics={tag.slug}",
```

Comment:
@csebianlander this is a change from the current behavior. Current topic tag linking behavior works like this:

- Take the page we are rendering the post preview for
- Go up in the tree until we find a filterable page type
- Use that as the base URL and append ?topics={tag.slug} to it

Are there any cases to support this complexity, versus just

- Take the current page we are rendering (which we know is a filterable page bc we don't use the post preview anywhere else) and append ?topics={tag.slug} to it

Reply:
no this seems fine!